query-node/mappings/src/common.ts

```diff
@@ -40,20 +41,36 @@ export function invalidMetadata(extraInfo: string, data?: unknown): void {
   logger.info(errorMessage, data)
 }
 
+/*
+  Creates a predictable and unique ID for the given content.
+*/
+export function createPredictableId(event: SubstrateEvent, content: string | Object): string {
+  const contentType = typeof content == 'string'
+    ? content
+    : JSON.stringify(content)
+
+  const id = `${event.blockNumber}_${event.index}_${contentType}`
```

Comment:
If the event index is relative to the transaction and not the block, this is not sufficiently deterministic if there happen to be two extrinsics in the same block that trigger an event at the same index with the same content.
If we go with this id implementation, we should probably add the transaction index as well.

Reply:
Id generation mechanism has been changed and is using simple number that is incremented each time (stored in db as `NextEntityId` entity).